autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/securestorage/encryption/EncryptionHelper.kt

```diff
@@ -53,12 +56,44 @@ interface EncryptionHelper {
 }
 
 @ContributesBinding(AppScope::class)
-class RealEncryptionHelper @Inject constructor() : EncryptionHelper {
+class RealEncryptionHelper @Inject constructor(
+    private val autofillFeature: AutofillFeature,
+) : EncryptionHelper {
     private val encryptionCipher = Cipher.getInstance(TRANSFORMATION)
     private val decryptionCipher = Cipher.getInstance(TRANSFORMATION)
 
+    private val encryptMutex = Mutex()
+    private val decryptMutex = Mutex()
+
+    override suspend fun encrypt(
+        raw: ByteArray,
+        key: Key,
+    ): EncryptedBytes {
+        return if (autofillFeature.createAsyncPreferences().isEnabled()) {
+            encryptAsync(raw, key)
+        } else {
+            encryptSync(raw, key)
+        }
+    }
```

Comment:
Given we're doing diskIO here, do we want to directly use dispatchers.io() here via `withContext` 
```suggestion
    override suspend fun encrypt(
        raw: ByteArray,
        key: Key,
    ): EncryptedBytes = withContext(dispatchers.io()) {
        return@withContext if (autofillFeature.createAsyncPreferences().isEnabled()) {
            encryptAsync(raw, key)
        } else {
            encryptSync(raw, key)
        }
    }
```